### PR TITLE
Refactor login

### PR DESCRIPTION
**Pull request by charlesBochet**
image

### CODE REVIEW

File: front/craco.config.js [hunk 1]
```diff
@@ -8,6 +8,9 @@ module.exports = {
           if (error.message === "ResizeObserver loop limit exceeded") {
             return false;
           }
+          if (error.message === "Unauthorized") {
```
Comment (charlesBochet) [inline front/craco.config.js:11]: this will remove the annoying "Unauthorized" overlay in dev

File: front/src/modules/auth/hooks/useAuth.ts [hunk 2]
```diff
@@ -97,16 +107,33 @@ export function useAuth() {
...
+
+      setCurrentUser(user);
+
+      return { user };
     },
-    [signUp, handleVerify],
+    [signUp, handleVerify, setCurrentUser],
   );
 
+  const googleLogin = useCallback((workspaceInviteHash?: string) => {
```
Comment (lucasbordeau) [inline front/src/modules/auth/hooks/useAuth.ts:121]: Better naming ?

Comment (lucasbordeau) [reply front/src/modules/auth/hooks/useAuth.ts:121]: handleGoogleLogin

File: front/src/modules/ui/input/components/TextInput.tsx [hunk 3]
```diff
@@ -117,16 +119,20 @@ export function TextInput({
...
     onBlur?.(e);
-    goBackToPreviousHotkeyScope();
+    if (!disableHotkeys) {
+      goBackToPreviousHotkeyScope();
+    }
   };
 
   useScopedHotkeys(
-    [Key.Enter, Key.Escape],
+    [Key.Escape],
```
Comment (lucasbordeau) [inline front/src/modules/ui/input/components/TextInput.tsx:135]: keep enter

File: front/src/modules/ui/layout/components/DefaultLayout.tsx [hunk 4]
```diff
@@ -38,22 +46,65 @@ type OwnProps = {
...
-  const currentUser = useRecoilState(currentUserState);
-  const userIsAuthenticated = !!currentUser;
+  const navigate = useNavigate();
+  const isMatchingLocation = useIsMatchingLocation();
+
+  const onboardingStatus = useOnboardingStatus();
+  useEffect(() => {
+    if (onboardingStatus === OnboardingStatus.OngoingUserCreation) {
+      if (
+        !isMatchingLocation(AppPath.SignIn) &&
```
Comment (lucasbordeau) [inline front/src/modules/ui/layout/components/DefaultLayout.tsx:56]: doubloon

File: front/src/modules/ui/layout/components/DefaultLayout.tsx [hunk 5]
```diff
@@ -38,22 +46,65 @@ type OwnProps = {
...
+      if (!isMatchingLocation(AppPath.CreateWorkspace)) {
+        navigate(AppPath.CreateWorkspace);
+      }
+    } else if (onboardingStatus === OnboardingStatus.OngoingProfileCreation) {
+      if (!isMatchingLocation(AppPath.CreateProfile)) {
+        navigate(AppPath.CreateProfile);
+      }
+    } else if (onboardingStatus === OnboardingStatus.Completed) {
+      if (
+        isMatchingLocation(AppPath.SignIn) ||
```
Comment (lucasbordeau) [inline front/src/modules/ui/layout/components/DefaultLayout.tsx:73]: doubloon

File: front/src/modules/ui/layout/components/DefaultLayout.tsx [hunk 6]
```diff
@@ -38,22 +46,65 @@ type OwnProps = {
...
 
 export function DefaultLayout({ children }: OwnProps) {
-  const currentUser = useRecoilState(currentUserState);
-  const userIsAuthenticated = !!currentUser;
+  const navigate = useNavigate();
+  const isMatchingLocation = useIsMatchingLocation();
+
+  const onboardingStatus = useOnboardingStatus();
+  useEffect(() => {
+    if (onboardingStatus === OnboardingStatus.OngoingUserCreation) {
```
Comment (lucasbordeau) [inline front/src/modules/ui/layout/components/DefaultLayout.tsx:54]: could we find a better naming for predicates ? maybe store it in variables ?

File: front/src/modules/users/components/UserProvider.tsx [hunk 7]
```diff
@@ -1,19 +1,14 @@
...
 export function UserProvider({ children }: React.PropsWithChildren) {
   const [, setCurrentUser] = useRecoilState(currentUserState);
   const [isLoading, setIsLoading] = useState(true);
 
-  const isLogged = useIsLogged();
-
-  const { data, loading } = useGetCurrentUserQuery({
-    skip: !isLogged,
-  });
+  const { data, loading } = useGetCurrentUserQuery({});
```
Comment (lucasbordeau) [inline front/src/modules/users/components/UserProvider.tsx:11]: {} useless ?

**Pull request merged**